In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
tat=pd.read_csv("LabTAT.csv")
tat

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

# Normality test

In [3]:
stats.shapiro(tat["Laboratory 1"])

ShapiroResult(statistic=0.990183413028717, pvalue=0.5507769584655762)

In [4]:
stats.shapiro(tat["Laboratory 2"])

ShapiroResult(statistic=0.9936320185661316, pvalue=0.8637352585792542)

In [5]:
stats.shapiro(tat["Laboratory 3"])

ShapiroResult(statistic=0.988633930683136, pvalue=0.42045730352401733)

In [6]:
stats.shapiro(tat["Laboratory 4"])

ShapiroResult(statistic=0.9913760423660278, pvalue=0.6619619131088257)

H0 is accepted, as Y1,Y2,Y3,Y4 are normal

# Variance

H0 == Variance of all 4 laboratories are same

H1 == Variance of all 4 laboratories are not same

Levene test is used for measuring the variances of k-variables.

In [7]:
stats.levene(tat["Laboratory 1"],tat["Laboratory 2"],tat["Laboratory 3"],tat["Laboratory 4"])

LeveneResult(statistic=2.599642500418024, pvalue=0.05161343808309816)

LeveneResult(statistic=2.599642500418024, pvalue=0.05161343808309816)

P-value of Variance test is == 0.051>significant level

# H0 is accepted
Thus we will perform one way ANOVA Test

# Model

H0 == Mean TAT for 4 laboratories is equal(There is no significant difference between TAT of the laboratories)

H1 == Mean TAT for 4 laboratories is not equal(There is a significant difference between TAT of the laboratories)

In [9]:
stats.f_oneway(tat["Laboratory 1"],tat["Laboratory 2"],tat["Laboratory 3"],tat["Laboratory 4"])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

### Thus H1 is accepted.
Mean TAT for 4 laboratories are not equal(There is a significant difference between TAT of the laboratories)